types/gensync/gensync-tests.ts

```diff
@@ -0,0 +1,161 @@
+import gensync = require('gensync');
+
+// Fake node fs.readFile for testing.
+declare function readFileCallback(
+    path: string,
+    encoding: 'utf8' | 'ascii',
+    cb: (err: Error, result: string) => void,
+): void;
+declare function readFileSync(path: string, encoding: 'utf8' | 'ascii'): string;
+declare function readFileAsync(path: string, encoding: 'utf8' | 'ascii'): Promise<string>;
+
+// $ExpectType Gensync<[path: string, encoding: "utf8" | "ascii"], string, unknown>
+const readFileFromSync = gensync({
+    name: 'readFile',
+    arity: 2,
+    sync: readFileSync,
+});
+
+// $ExpectType (path: string, encoding: "utf8" | "ascii") => string
```

Comment:
Normally I'd like to avoid ExpectType, but for these sorts of tests, what I'm really testing is that the parameter names are preserved, which leads to better editor functionality and error messages.